core/domain/story_services.py

```diff
@@ -613,3 +617,247 @@ def record_completed_node_in_story_context(user_id, story_id, node_id):
+    """
+
+    model = story_models.StoryRightsModel.get(story_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_story_rights_from_model(model)
+
+
+def check_can_edit_story(user, story_rights):
+    """Checks whether the user can edit the given story.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        story_rights: StoryRights or None. Rights object for the given story.
+
+    Returns:
+        bool. Whether the given user can edit the given story.
+    """
+    if story_rights is None:
+        return False
+    if role_services.ACTION_EDIT_ANY_STORY in user.actions:
+        return True
+    if role_services.ACTION_EDIT_OWNED_STORY not in user.actions:
+        return False
+    if story_rights.is_manager(user.user_id):
+        return True
+
+    return False
+
+
+def assign_role(committer, assignee, new_role, story_id):
+    """Assigns a new role to the user.
+
+    Args:
+        committer: UserActionsInfo. UserActionsInfo object for the user
+            who is performing the action.
+        assignee: UserActionsInfo. UserActionsInfo object for the user
+            whose role is being changed.
+        new_role: str. The name of the new role. Possible values are:
+            ROLE_MANAGER
+        story_id: str. ID of the story.
+
+    Raises:
+        Exception. The committer does not have rights to modify a role.
+        Exception. The assignee is already a manager for the story.
+        Exception. The assignee doesn't have enough rights to become a manager.
+        Exception. The role is invalid.
+    """
+    committer_id = committer.user_id
+    story_rights = get_story_rights(story_id)
+    if (role_services.ACTION_MODIFY_ROLES_FOR_ANY_ACTIVITY not in
+            committer.actions):
+        logging.error(
+            'User %s tried to allow user %s to be a %s of story %s '
+            'but was refused permission.' % (
+                committer_id, assignee.user_id, new_role, story_id))
+        raise Exception(
+            'UnauthorizedUserException: Could not assign new role.')
+
+    assignee_username = user_services.get_username(assignee.user_id)
+    if role_services.ACTION_EDIT_OWNED_STORY not in assignee.actions:
+        raise Exception(
+            'The assignee doesn\'t have enough rights to become a manager.')
+
+    old_role = story_domain.ROLE_NONE
+    if story_rights.is_manager(assignee.user_id):
+        old_role = story_domain.ROLE_MANAGER
+
+    if new_role == story_domain.ROLE_MANAGER:
+        if story_rights.is_manager(assignee.user_id):
+            raise Exception('This user already is a manager for this story')
+        story_rights.manager_ids.append(assignee.user_id)
+    elif new_role == story_domain.ROLE_NONE:
+        if story_rights.is_manager(assignee.user_id):
+            story_rights.manager_ids.remove(assignee.user_id)
+        else:
+            old_role = story_domain.ROLE_NONE
```

Comment:
This doesn't make sense, I think -- why would we be changing NONE to NONE? Seems like an error case...

(Make sure you have tests for this case.)

Reply:
I must've missed it. Thanks, removed it.